packages/containers/src/ObjectViewer/ObjectViewer.container.js

```diff
@@ -53,18 +61,29 @@ class ObjectViewer extends React.Component {
 
 	constructor(props) {
 		super(props);
-		this.onClick = this.onClick.bind(this);
+		this.onToggle = this.onToggle.bind(this);
+		this.onEdit = this.onEdit.bind(this);
 		this.onChange = this.onChange.bind(this);
+		this.onSelect = this.onSelect.bind(this);
 	}
 
-	onClick(event, data) {
+	onToggle(event, data) {
 		let newState;
 		if (data.isOpened) {
 			newState = close(data.jsonpath, this.props.state);
 		} else if (data.isOpened === false) {
 			// we don't want to match on undefined as false
 			newState = open(data.jsonpath, this.props.state);
-		} else if (data.edit === false) {
+		}
+
+		if (newState) {
+			this.props.setState(newState);
```

Comment:
since this use a previous state to compute the new one, use the functionnal way to setState
the whole upper block should fit in this function

https://facebook.github.io/react/docs/react-component.html#setstate
this.setState((prevState, props) => {
  return {counter: prevState.counter + props.step};
});

Reply:
prevState is not type consistent between Map and plain old js object which causes functionnal way of setState to fail while using defined operation function (select, open...) that are using map.set method
+ code consistency with other method operation functions